### PR TITLE
Authenticating non azure openai

### PR DESCRIPTION
**Pull request by Njeriiii**
What does this implement/fix? Explain your changes.
---------------------------------------------------
 This class is a LocalInspectionTool that checks if the endpoint method is used with KeyCredential for non-Azure OpenAI clients.
 If the endpoint method is used with KeyCredential for non-Azure OpenAI clients, a warning is registered.

Has this been tested?
---------------------------
- [x] Tested

Any relevant logs, screenshots, error output, etc.?
-------------------------------------
image

### CODE REVIEW

**Review by mssfang — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/README.md [hunk 1]
```diff
@@ -211,4 +214,17 @@ integration, telemetry connectivity, and Azure Toolkit integration.
...
+  actions in a single request.
+
+11. #### Authenticating a Non-Azure OpenAI Client with KeyCredential
+
+- **Anti-pattern**: Assigning the endpoint value when creating a Non-Azure OpenAI client using the KeyCredential in
+  .credential(KeyCredential).
+- **Issue**: KeyCredential is the only required parameter in `.credential(KeyCredential)` for authenticating requests to
+  non-Azure OpenAI APIs.
+- **Severity: WARNING**
+- **Recommendation**: Omit Endpoint: Do not specify the endpoint parameter unless you are working with Azure-specific
```
Comment (samvaity) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/README.md]: **Recommendation**: Omit Endpoint: ?

Not sure I understand the recommendation.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/resources/META-INF/azure-intellij-plugin-azure-sdk.xml [hunk 2]
```diff
@@ -168,5 +168,19 @@
                 com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool.DetectDiscouragedClientCheck
             </class>
         </localInspection>
+
+        <localInspection
+                language="JAVA"
+                shortName="EndpointOnNonAzureOpenAIAuthCheck"
+                displayName="Endpoint On Non-Azure OpenAI Auth Check"
```
Comment (samvaity) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/resources/META-INF/azure-intellij-plugin-azure-sdk.xml]:
```suggestion
                displayName="Endpoint On Non-Azure OpenAI Authentication Check"
```
We should try avoiding abbreviations in naming/descriptions.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/EndpointOnNonAzureOpenAIAuthCheckTest.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+     * creates a JavaElementVisitor object for visiting method call expressions
+     */
+    private JavaElementVisitor createVisitor() {
+        return new EndpointOnNonAzureOpenAIAuthVisitor(mockHolder);
+    }
+
+    /**
+     * This method verifies if the registerProblem method is called with the correct parameters
+     */
+    private void verifyRegisterProblem(int numOfInvocation, String endpoint, String credential, String KeyCredentialPackageName, String azurePackageName) {
```
Comment (samvaity) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/EndpointOnNonAzureOpenAIAuthCheckTest.java]: Casing

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/EndpointOnNonAzureOpenAIAuthCheckTest.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+     * This test checks if the endpoint method is used with KeyCredential but for Azure OpenAI clients.
+     * If the endpoint method is used with KeyCredential but for Azure OpenAI clients, no warning is registered.
+     */
+    @Test
+    public void testWithAzureKeyCredential() {
+
+        int numOfInvocation = 0;
+        String endpoint = "endpoint";
+        String credential = "notCredential";
+        String KeyCredentialPackageName = "com.azure.core.credential.AzureKeyCredential";
```
Comment (samvaity) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/EndpointOnNonAzureOpenAIAuthCheckTest.java]: update to camel casing

**Review by samvaity — APPROVED** [review]
Left some comments, but lgtm otheriwse.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/README.md (hunk 1 above)
Comment (Njeriiii) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/README.md]: .endpoint shouldn't be in the method chain when authenticating non-azure open ai clients. I'll try to make it more descriptive.

**Pull request merged**